componentLibraries/defaultLibrary/Connectivity/FMIWrapper.hpp

```diff
@@ -0,0 +1,1324 @@
+                 return;
+             }
+     
+             //Write outputs
+             std::map<fmi3ValueReference,double*>::iterator it;
+             for(it = mFloat64Outputs.begin(); it != mFloat64Outputs.end(); it++) {
+                 fmi3Float64 value;
+                 status = fmi3GetFloat64(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mFloat32Outputs.begin(); it != mFloat32Outputs.end(); it++) {
+                 fmi3Float32 value;
+                 status = fmi3GetFloat32(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mInt64Outputs.begin(); it != mInt64Outputs.end(); it++) {
+                 fmi3Int64 value;
+                 status = fmi3GetInt64(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mInt32Outputs.begin(); it != mInt32Outputs.end(); it++) {
+                 fmi3Int32 value;
+                 status = fmi3GetInt32(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mInt16Outputs.begin(); it != mInt16Outputs.end(); it++) {
+                 fmi3Int16 value;
+                 status = fmi3GetInt16(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mInt8Outputs.begin(); it != mInt8Outputs.end(); it++) {
+                 fmi3Int8 value;
+                 status = fmi3GetInt8(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mUInt64Outputs.begin(); it != mUInt64Outputs.end(); it++) {
+                 fmi3UInt64 value;
+                 status = fmi3GetUInt64(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mUInt32Outputs.begin(); it != mUInt32Outputs.end(); it++) {
+                 fmi3UInt32 value;
+                 status = fmi3GetUInt32(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mUInt16Outputs.begin(); it != mUInt16Outputs.end(); it++) {
+                 fmi3UInt16 value;
+                 status = fmi3GetUInt16(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mUInt8Outputs.begin(); it != mUInt8Outputs.end(); it++) {
+                 fmi3UInt8 value;
+                 status = fmi3GetUInt8(fmu, &it->first, 1, &value, 1);
+                 (*it->second) = (double)value;
+             }
+             for(it = mBoolOutputs.begin(); it != mBoolOutputs.end(); it++) {
+                 bool temp;
+                 status = fmi3GetBoolean(fmu, &it->first, 1, &temp, 1);
+                 (*it->second) = temp;
+             }
+         }
+    }
```

Comment:
I will be honest, I did not thoroughly review all of the repetitive code for all the types, but it seems like you got them all right as far as I can see

Reply:
I have already discovered a few mistakes. It is har to check the functionality, since most FMUs only use a small subset of all features. We will need to fix more bugs as they are discovered.